src/wagtail_personalisation/utils.py

```diff
@@ -116,3 +116,11 @@ def can_delete_pages(pages, user):
         if not variant.permissions_for_user(user).can_delete():
             return False
     return True
+
+
+def get_client_ip(request):
+    try:
+        x_forwarded_for = request.META['HTTP_X_FORWARDED_FOR']
+        return x_forwarded_for.split(',')[-1].strip()
```

Comment:
I wonder if we should make it configurable? In some cases user's IP will be second to the last. For example, google cloud load balancer puts proxies at the end.

Reply:
We probably should, yes.

Reply:
I've changed this. Can you re-review please?